scripts/src/pullrequest/prartifact.py

```diff
@@ -10,33 +10,79 @@
 sys.path.append('../')
 from checkprcontent import checkpr
 
+pr_files = []
+pr_labels = []
+xRateLimit = "X-RateLimit-Limit"
+xRateRemain = "X-RateLimit-Remaining"
+
 # TODO(baijum): Move this code under chartsubmission.chart module
 def get_modified_charts(api_url):
-    files_api_url = f'{api_url}/files'
-    headers = {'Accept': 'application/vnd.github.v3+json'}
-    r = requests.get(files_api_url, headers=headers)
+    files = get_modified_files(api_url)
     pattern,_ = checkpr.get_file_match_compiled_patterns()
-    for f in r.json():
-        print(f"[DEBUG] TYPE OF F: {type(f)}")
-        print(f"[DEBUG] VALUE OF F: {f}")
-        m = pattern.match(f["filename"])
-        if m:
-            category, organization, chart, version = m.groups()
+    for file in files:
+        match = pattern.match(file)
+        if match:
+            category, organization, chart, version = match.groups()
             return category, organization, chart, version
 
     return "", "", "", ""
 
 def get_modified_files(api_url):
-    files_api_url = f'{api_url}/files'
-    headers = {'Accept': 'application/vnd.github.v3+json'}
-    r = requests.get(files_api_url, headers=headers)
-    pr_files = []
-    print(f"[INFO] file info in PR {r.json()}")
-    for f in r.json():
-        if "filename" in f:
-            pr_files.append(f["filename"])
+    if not pr_files:
+        page_number = 1
+        max_page_size,page_size = 100,100
+        headers = {'Accept': 'application/vnd.github.v3+json','Authorization': f'Bearer {os.environ.get("BOT_TOKEN")}'}
+        files_api_url = f'{api_url}/files'
+
+        while page_size == max_page_size:
+
+            files_api_query = f'{files_api_url}?per_page={page_size}&page={page_number}'
+            print(f"[INFO] Query files : {files_api_query}")
+            r = requests.get(files_api_query,headers=headers)
+            files = r.json()
+            page_size = len(files)
+            page_number += 1
+
+            if xRateLimit in r.headers:
+                print(f'[DEBUG] {xRateLimit} : {r.headers[xRateLimit]}')
+            if xRateRemain in r.headers:
+                print(f'[DEBUG] {xRateRemain}  : {r.headers[xRateRemain]}')
+
+            if "message" in files:
+                print(f'[ERROR] getting pr files: {files["message"]}')
+                sys.exit(1)
+            else:
+                for file in files:
+                    if "filename" in file:
+                        pr_files.append(file["filename"])
+                    if "message" in file:
```

Comment:
I think this condition check is dead code here, since we are already checking earlier if `"message" in files` and exiting if it is true.

Reply:
Good spot, removed it